restapi_test.go

```diff
@@ -235,3 +237,41 @@ func Test_unmarshal(t *testing.T) {
 		t.Errorf("Unexpected error type: %T", err)
 	}
 }
+
+func TestWithContext(t *testing.T) {
+	// Set up a test context.
+	type key struct{}
+	ctx := context.WithValue(context.Background(), key{}, "foo")
+
+	// Set up a test client.
+	session, err := New("")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testErr := errors.New("test")
+
+	// Intercept the request to assert the context.
+	session.Client.Transport = roundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		val, _ := r.Context().Value(key{}).(string)
+		if val != "foo" {
```

Comment:
In relation to https://github.com/bwmarrin/discordgo/pull/1337/files#r1131680827
```suggestion
		if val != "value" {
```